src/nvim/main.c

```diff
@@ -802,6 +806,97 @@ static void init_locale(void)
 }
 #endif
 
+/// Handle remote subcommands
+static void handle_remote_client(mparm_T *params, int remote_args,
+                                 char *server_addr, int argc, char **argv)
+{
+    Object rvobj = OBJECT_INIT;
+    rvobj.data.dictionary = (Dictionary)ARRAY_DICT_INIT;
+    rvobj.type = kObjectTypeDictionary;
+    CallbackReader on_data = CALLBACK_READER_INIT;
+    const char *error = NULL;
+    uint64_t rc_id = 0;
+    if (server_addr != NULL) {
+      rc_id = channel_connect(false, server_addr, true, on_data, 50, &error);
+    }
+    if (error) {
```

Comment:
I'd place this within the previous if as it only is relevant for `channel_connect()`

Reply:
Good call!

I realized when looking at this that it should also respect the silent flag, so I ended up passing the error into Lua to let it decide if and how to report.